backend/notifications.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import time
+
+
+def HandleNotification(firebase, notification_id, notification):
+  """Helper function to propogate a notification."""
+
+  # TODO: Send to ionic
+  if 'playerId' in notification:
+    players = [notification['playerId']]
+  elif 'groupId' in notification:
+    players = firebase.get('/groups/%s' % notification['groupId'],
+                           'players')
+  else:
+    logging.error('Notification %s does not have a playerId or a groupId!' % (
+        notification_id))
+    return
+
+  for player in players:
+    firebase.put('/players/%s/notifications' % player,
+                  notification_id, notification)
+
+
+def ExecuteNotifications(request, firebase):
+  """INTERNAL ONLY: Send the notifications.
+
+  This method will execute queued notifications and send them to apps and add
+  them to the player specific notification database URLs.
+  """
+  # Handle notifications, 20 at a time so we don't need to get the entire
+  # notification database.
+  get_params = {
+      'orderBy': '"sent"',
```

Comment:
Yes, it needs to be a valid json string. https://firebase.google.com/docs/database/rest/retrieve-data#section-complex-queries